oat-sa/tao-core
# Feature/AUT-3632/Apply the Solar Design to the forms

### Related to: https://oat-sa.atlassian.net/browse/AUT-3646

### Summary

Apply the Solar Design to the forms
![image](https://github.com/oat-sa/tao-core/assets/1500098/0902b38e-d18a-4037-b3a9-b8e039c99337)
![image](https://github.com/oat-sa/tao-core/assets/1500098/72999444-b1f8-4f2b-935a-b926fe7b25cc)



### Details

To activate the feature, you need the feature flag `FEATURE_FLAG_SOLAR_DESIGN_ENABLED`.
```sh
php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true
```

Once activated, the CSS class `solar-design` is added to the body element of the page. Then, the stylesheet automatically applies.

The Solar Design has been partially replicated in the CSS variables. For compatibility reasons, some have been simplified.

Most of the forms are now following the Solar Design. A few additions will come soon either in another PR or in a next ticket. 

### How to test

- check out the branch: `git checkout -t origin/feature/AUT-3632/solar-design-forms`
- activate the feature flag: `php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true`
- log in to TAO backoffice
- look at the forms, compare with the UX mockups
